tox.ini

```diff
@@ -0,0 +1,113 @@
+# Copyright 2024 Kmonas Sotirchos
+# See LICENSE file for licensing details.
+
+[tox]
+no_package = True
+skip_missing_interpreters = True
+env_list = format, lint, static, unit
+min_version = 4.0.0
+
+[vars]
+src_path = {tox_root}/src
+tests_path = {tox_root}/tests
+all_path = {[vars]src_path} {[vars]tests_path}
+
+[testenv]
+allowlist_externals = poetry, which
+deps=
+    poetry
+set_env =
+    PYTHONPATH = {tox_root}/lib:{[vars]src_path}
+    PYTHONBREAKPOINT=pdb.set_trace
+    PY_COLORS=1
+pass_env =
+    PYTHONPATH
+    CHARM_BUILD_DIR
+    MODEL_SETTINGS
+
+[testenv:update-requirements]
+description = Apply coding style standards to code
+deps =
+    poetry
+commands =
+    python --version
+    which python
+    poetry config warnings.export false
+    poetry lock
+    poetry install --no-root
+    poetry export -f requirements.txt -o requirements.txt --without-hashes
+
```

Comment:
Adding some context from a sync with @orfeas-k, which we will also put in a small informational spec:

1. We'll stick with having a `requirements.txt` file committed, since
    1. We'd like users to be able to use `pip`, to install deps, and not rely on poetry
    2. We see issues in mysql when the `requirements.txt` is missing [[1]](https://github.com/canonical/mysql-k8s-operator/blob/45db15d98819a70d6ff9c7a52a5961ec4c51dfcb/.github/workflows/release.yaml#L21) [[2]]([[2]](https://github.com/canonical/mysql-k8s-operator/blob/45db15d98819a70d6ff9c7a52a5961ec4c51dfcb/charmcraft.yaml#L18)) and an ongoing discussion https://github.com/canonical/charmcraft/issues/1389#issuecomment-2370521473, and we'd like to be this much in the bleeding edge
2. Have a `tox -e update-requirements` that updates both `poetry.lock` and `requirements.txt`

Reply:
For the above, we saw that poetry had a warning that `poetry export` won't be installed in the future.

To get around this we:
1. Created a specific group for `update-requirements`
2. Included the `poetry-plugin-export` package
3. Silenced the specific warning for the export

14e0889